### ISSUE TITLE
[Bug]: It seems that the QueryNode deleteBuffer entriesNum is not reset after processing a delete.

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#37057]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.4-20241022-55caf411-amd64
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. server config
```
config:
    dataCoord:
      compaction:
        taskPrioritizer: level
      enableActiveStandby: true
      segment:
        expansionRate: 1.15
        maxSize: 2048
        sealProportion: 0.12
    indexCoord:
      enableActiveStandby: true
    log:
      level: debug
    queryCoord:
      enableActiveStandby: true
    queryNode:
      **levelZeroForwardPolicy: RemoteLoad
      streamingDeltaForwardPolicy: Direct**
    quotaAndLimits:
      dml:
        deleteRate:
          max: 0.5
        enabled: true
        insertRate:
          max: 8
      limitWriting:
        deleteBufferRowCountProtection:
          enabled: true
          highWaterLevel: 10000000
          lowWaterLevel: 1000000
        deleteBufferSizeProtection:
          enabled: true
          highWaterLevel: 536870912
          lowWaterLevel: 134217728
        growingSegmentsSizeProtection:
          enabled: true
          highWaterLevel: 0.2
          lowWaterLevel: 0.1
          minRateRatio: 0.5
        l0SegmentsRowCountProtection:
          enabled: true
          highWaterLevel: 10000000
          lowWaterLevel: 1000000
        memProtection:
          dataNodeMemoryHighWaterLevel: 0.9
          dataNodeMemoryLowWaterLevel: 0.85
          queryNodeMemoryHighWaterLevel: 0.9
          queryNodeMemoryLowWaterLevel: 0.85
      limits:
        complexDeleteLimitEnable: true
```
2. Try to delete 7479634 out of 77,300,000 entities by expr `'id > "u"'` at 2024/10/22 06:34:17.961(UTC), actually only 241978 is deleted. Client gets memory force deny error:
```
RPC error: [delete], <MilvusException: (code=9, message=quota exceeded[reason=memory quota exceeded, please allocate more resources])>
```
but I think the real reason is  deleteBufferRowCountProtection
![image](https://github.com/user-attachments/assets/cdb28509-ef9e-4fbd-a2fe-eb73626a5d1d)
- querynode mem usage
![image](https://github.com/user-attachments/assets/a1fee905-ed07-46cf-8fa4-df720cb24768)

3. refresh config
```
compact-opt-rate-100m-1/config/quotaandlimits.limitwriting.deleteBufferRowCountProtection.highWaterLevel
50000000
compact-opt-rate-100m-1/config/quotaandlimits.limitwriting.deleteBufferRowCountProtection.lowWaterLevel
5000000
compact-opt-rate-100m-1/config/quotaandlimits.limitwriting.deleteBufferSizeProtection.highWaterLevel
1073741824
compact-opt-rate-100m-1/config/quotaandlimits.limitwriting.deleteBufferSizeProtection.lowWaterLevel
536870912
compact-opt-rate-100m-1/config/quotaandlimits.limitwriting.l0SegmentsRowCountProtection.highWaterLevel
50000000
compact-opt-rate-100m-1/config/quotaandlimits.limitwriting.l0SegmentsRowCountProtection.lowWaterLevel
5000000
```
4. delete by expr `'id > "u"'` again at 2024-10-22 07:20:35(UTC), client gets same memory quota exceed error.From the server QuotaCenter logs we can see the deletebuffer entriesNum is still 249963370. And the delete message is handled by queryNode yet.
![image](https://github.com/user-attachments/assets/372c1fcd-cffe-4530-97d3-269623087b19)
![image](https://github.com/user-attachments/assets/8b945a8a-6d78-4b9a-bd4d-83942a765f65)

# metrics: 
[metrics link](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=db7e3557-d29c-4d2b-be40-9f465bcec112&var-namespace=qa-milvus&var-instance=compact-opt-rate-100m-1&var-collection=All&var-app_name=milvus&from=1729578524000&to=1729582813000)

### Expected Behavior

1. client gets force deny error about delete buffer
2. increase deleteBufferRowCountProtection warter level, no delete force deny

### Milvus Log

pods:
```
compact-opt-rate-100m-1-milvus-datanode-85cdc697f9-7k7m9          1/1     Running     0               158m    10.104.33.72    4am-node36   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-85cbbb7c98-jp5rv         1/1     Running     0               167m    10.104.13.41    4am-node16   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-85cbbb7c98-js2fq         1/1     Running     0               166m    10.104.20.185   4am-node22   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-85cbbb7c98-mx9hq         1/1     Running     0               168m    10.104.15.184   4am-node20   <none>           <none>
compact-opt-rate-100m-1-milvus-mixcoord-78ffd69f7c-rxvjk          1/1     Running     0               165m    10.104.15.189   4am-node20   <none>           <none>
compact-opt-rate-100m-1-milvus-proxy-74f599fdcf-xkn9z             1/1     Running     0               156m    10.104.33.74    4am-node36   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-676cc6b566-2t8kl       1/1     Running     0               162m    10.104.20.187   4am-node22   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-676cc6b566-5xvqj       1/1     Running     0               160m    10.104.23.158   4am-node27   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-676cc6b566-cdzwf       1/1     Running     0               163m    10.104.15.191   4am-node20   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-676cc6b566-h7qb8       1/1     Running     0               164m    10.104.30.101   4am-node38   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-676cc6b566-qfv6n       1/1     Running     0               159m    10.104.18.250   4am-node25   <none>           <none>
```

**Comment by congqixia** [comment]
@ThreadDao patch merged, could you please verify

**Comment by ThreadDao** [comment]
fixed `2.4-20241023-2abc28ec-amd64`
